lib/mp4/transmuxer.js

```diff
@@ -546,51 +547,103 @@ Transmuxer = function(options) {
     videoTrack,
     audioTrack,
     packetStream, parseStream, elementaryStream,
-    aacStream, h264Stream,
+    adtsStream, h264Stream,aacStream,
     videoSegmentStream, audioSegmentStream, captionStream,
-    coalesceStream;
+    coalesceStream,
+    aacfile;
 
-  Transmuxer.prototype.init.call(this);
-  options = options || {};
+  this.setupAACPipeline = function() {
+    this.metadataStream = new AacStream.MetadataStream();
+    options.metadataStream = this.metadataStream;
 
-  this.baseMediaDecodeTime = options.baseMediaDecodeTime || 0;
+    // set up the parsing pipeline
+    aacStream = new AacStream.AacStream();
+    adtsStream = new AdtsStream();
 
-  // expose the metadata stream
-  this.metadataStream = new m2ts.MetadataStream();
-
-  options.metadataStream = this.metadataStream;
-
-  // set up the parsing pipeline
-  packetStream = new m2ts.TransportPacketStream();
-  parseStream = new m2ts.TransportParseStream();
-  elementaryStream = new m2ts.ElementaryStream();
-  aacStream = new AacStream();
-  h264Stream = new H264Stream();
-  captionStream = new m2ts.CaptionStream();
-  coalesceStream = new CoalesceStream(options);
-
-  // disassemble MPEG2-TS packets into elementary streams
-  packetStream
-    .pipe(parseStream)
-    .pipe(elementaryStream);
-
-  // !!THIS ORDER IS IMPORTANT!!
-  // demux the streams
-  elementaryStream
-    .pipe(h264Stream);
-  elementaryStream
-    .pipe(aacStream);
-
-  elementaryStream
-    .pipe(this.metadataStream)
-    .pipe(coalesceStream);
-
-  // Hook up CEA-608/708 caption stream
-  h264Stream.pipe(captionStream)
-    .pipe(coalesceStream);
+    //audioSegmentStream = new AudioSegmentStream();
 
-  // hook up the segment streams once track metadata is delivered
-  elementaryStream.on('data', function(data) {
+    coalesceStream = new CoalesceStream(options);
+
+    aacStream.pipe(adtsStream);
+    aacStream.pipe(this.metadataStream);
+    this.metadataStream.pipe(coalesceStream);
+
+    this.metadataStream.on('timestamp', aacStream.setTimestamp.bind(aacStream));
+    this.addAACListener();
+
+  };
+
+
+  this.addAACListener = function() {
+      aacStream.on('data', function(data) {
+    var i;
+    // hook up the video segment stream to the first track with h264 data
+
+    if (data.type === 'timed-metadata') {
+      //i = data.tracks.length;
+      var track = {
+              timelineStartInfo: {
+                baseMediaDecodeTime: 0
+              },
+              codec: 'adts',
+              type: 'audio'
+            };
+
+      if (track && !audioSegmentStream) {
+        // hook up the audio segment stream to the first track with aac data
+        coalesceStream.numberOfTracks++;
+        audioSegmentStream = new AudioSegmentStream(track);
+
+        // Set up the final part of the audio pipeline
+        adtsStream
+          .pipe(audioSegmentStream)
+          .pipe(coalesceStream);
+
```

Comment:
There are extra empty newlines in this file... is this where all the ones that were missing in the other file snuck off to? :smile: